factory/crypto/cryptoComponents.go

```diff
@@ -334,14 +329,16 @@ func (ccf *cryptoComponentsFactory) createCryptoParams(
 
 		return ccf.generateCryptoParams(keygen, "in import mode", handledPrivateKeys)
 	}
-	if ccf.noKeyProvided {
-		return ccf.generateCryptoParams(keygen, "with no-key flag enabled", make([][]byte, 0))
-	}
-	if len(handledPrivateKeys) > 0 {
-		return ccf.generateCryptoParams(keygen, "running with a provided allValidatorsKeys.pem", handledPrivateKeys)
+	cp, err := ccf.readCryptoParams(keygen)
+	if err == nil {
+		cp.handledPrivateKeys = handledPrivateKeys
```

Comment:
with this new approach, if `validatorKey.pem` is present, we'll use it as the **observer's** key that manages all keys

may be confusing, perhaps we should update the `Usage` field of the `validator-key-pem-file` flag?

Reply:
updated the Usage section.